### ISSUE TITLE
Card with empty title breaks Dashboards

### ISSUE THREAD

**Issue by freakinruben** [metabase/metabase#11163]
**Describe the bug**
I'm experiencing a weird bug with 2 specific SQL questions already existing on a dashboard that has been around since February. 

I currently can't make any changes to the dashboard, pressing "save" results in the button changing to "save failed". 

> Side note: if I remove a filter from the dashboard and the "save failed", the filter is actually removed so it didn't roll back..

I narrowed the failed save on the dashboard down to 2 specific questions, which, when they are saved with the table visualisation, don't cause any problems (I can save the dashboard), but when they are bar-charts the dashboard save fails.

I tested this by making a new dashboard, adding one of the 2 problematic questions and could reproduce the "save failed" in bar-chart mode.
Resizing, and dragging the broken question also doesn't work properly, it doesn't register the release of the mouse-button after a resize so I can keep dragging the element around.

The moment I press "edit" on the dashboard, the following error appeared in the js-console:
`TypeError: t[0] is undefined` (`app-main.bundle.js:5:521779`) 
The stack-trace is just a bunch of react render-cycles

I then recreated one of the problematic questions, including all it's field filters (11) and tried a new dashboard with this question. No problems whatsoever.

I compared the records of the old and new question in the metabase dashboard and the values of 2 columns where different:

- `dataset query`: no significant difference in the json. Only the `id` of the field-filters was different ofcourse and the order of the keys in the json slightly differed. 
- `visualization_setting`: differed quite a bit, I've probably tried out different visuals on the broken question:
  - old: 

```json
{"funnel.metric":"c","card.title":"","graph.dimensions":["complaint"],"graph.metrics":["count"],"table.pivot_column":"count","table.cell_column":"complaint","series_settings":{"count":{"display":"bar","color":"#509EE3"}}}
```

- new: 

```json
{"graph.dimensions":["complaint"],"graph.metrics":["count"]}
```

**Logs**
I compared the js console output on a new dashboard with a broken question and with the replicated question/
![Screenshot 2019-10-17 at 13 29 52](https://user-images.githubusercontent.com/326344/67008078-0b59ee00-f0e9-11e9-93f5-a4c3e92202e3.png)

From the `troubleshooting log`:
```
[f9786fc0-cc47-44f1-9a93-1722d567ac3a] 2019-10-17T14:31:59+02:00 DEBUG metabase.middleware.log PUT /api/dashboard/75 200 10.0 ms (5 DB calls) Jetty threads: 3/50 (7 idle, 0 queued) (263 total active threads) Queries in flight: 1
[f9786fc0-cc47-44f1-9a93-1722d567ac3a] 2019-10-17T14:31:59+02:00 DEBUG metabase.middleware.log GET /api/dashboard/75 200 14.4 ms (10 DB calls) Jetty threads: 3/50 (7 idle, 0 queued) (263 total active threads) Queries in flight: 1
```

**To Reproduce**
I rely have no idea what's triggering the bug, but could it potentially have to do with the `visualization_setting`?

**Expected behavior**
Being able to edit the dashboard that contains these 2 questions.

**Current work around**

1. Save the 2 problematic questions using a table visualisation.
2. Edit the dashboard
3. Save the dashboard
4. Chang the 2 problematic questions back to bar-charts

**Screenshots**

**Information about your Metabase Installation:**

You can get this information by going to Admin -> Troubleshooting.

- Your browser and the version: Firefox 69.0.2
- Your operating system: OS X 10.14
- Your databases: Postgres
- Metabase version: 0.33.4
- Metabase hosting environment: Docker
- Metabase internal database: Postgres

**Severity**
> How severe an issue is this bug to you? Is this annoying, blocking some users, blocking an upgrade or blocking your usage of Metabase entirely?
> Note: the more honest and specific you are here the more we will take you seriously.

Extremely annoying but able to work around.

To be honest, not rolling back on a failed save is a lot more severe

**Comment by flamber** [comment]
Hi @freakinruben 
Interesting. This is one of those difficult-to-reproduce issues. I think it might be related to #10796.

As for the removing of a filter, clicking Save, and it saying "Save failed", but it still actually saves, is definitely not something that should happen.
But looking at your log, it actually saves - that's the `PUT /api/dashboard/75 200`
Do you also have a log of when it fails?

Is it possible for you to remove the two questions from the dashboard and save - and then add them again?

**Comment by sarahjane1987** [comment]
@freakinruben I seem to be experiencing the same issue. I've copied your text below where it's the same for me.

**Describe the bug**
I am unable to save, drop after dragging, remove any charts from my dashboard, and filters are not working correctly. It seems to be tied to 1 or 2 corrupt visualizations. Everything was working fine until those were added.

I also currently can't make any changes to the dashboard, pressing "save" results in the button changing to "save failed".

The issues seemed to start when I changed the visualizations to line graphs. They had been on the dashboard as bar charts up until then with no issue. 

I tested this by making a new dashboard and adding both of the 2 problematic questions and could reproduce the "save failed" 
Resizing, and dragging the broken question also doesn't work properly, it doesn't register the release of the mouse-button after a resize so I can keep dragging the element around.

**To Reproduce**
I rely have no idea what's triggering the bug, but could it potentially have to do with the visualization_setting?

**Expected behavior**
Being able to edit the dashboard that contains these 2 questions.

**Current work around**
None. I can't save the dashboard or remove the problematic visualizations.

**Information about your Metabase Installation:**

You can get this information by going to Admin -> Troubleshooting.
{
  "browser-info": {
    "language": "en-US",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.5+10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.5",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.5+10",
    "os.name": "Linux",
    "os.version": "4.14.152-98.182.amzn1.x86_64",
    "user.language": "en",
    "user.timezone": "US/Eastern"
  },
  "metabase-info": {
    "databases": [
      "mysql",
      "h2"
    ],
    "hosting-env": "elastic-beanstalk",
    "application-database": "postgres",
    "run-mode": "prod",
    "version": {
      "date": "2020-01-13",
      "tag": "v0.34.1",
      "branch": "release-0.34.x",
      "hash": "265695c"
    },
    "settings": {
      "report-timezone": "US/Eastern"
    }
  }
}
**Severity**
Blocking my use of this dashboard entirely.

FYI @likeshumidity

**Comment by dacort** [comment]
@freakinruben Hey there 👋 – I noticed when looking at your `visualization_settings` data that your `card.title` is set to an empty string. On a hunch, I made a simple dashboard with two line charts. I then edited the dashboard, clicked into the settings dialog of one of the cards, and deleted the Title. I then hit the "Save" button and got the dreaded "Save failed!" 😱. Then if I went back to edit the dashboard, I hit all these wonky issues. So this looks like we may not be gracefully handling an empty string for the card title.

@sarahjane1987 Do you know if you had the same scenario with an empty card title? Are you able to include the contents of the `visualization_settings` field from your Metabase application database? 

I was able to workaround this by connecting directly to the application database and updating the `card.title` property in the `visualization_settings` column:

```sql
UPDATE report_dashboardcard
SET visualization_settings = '{"graph.dimensions":["STATE"],"graph.metrics":["COUNT"],"series_settings":{"COUNT":{"display":"line"}},"card.title":"oops"}'
WHERE id = 2;
```

If you want no card title, I was also able to get that to work by setting the title to a "space" character (` `) and that can be done in the UI itself.

**Comment by sarahjane1987** [comment]
@dacort Yes! I do have the same! I set one of them to no title because I wanted to set a longer title as a header and didn't want two titles.

Thanks for cracking the case!!

**Comment by dacort** [comment]
<div>Agent Damon Cortesi linked Freshdesk ticket <a href="https://metabase.freshdesk.com/a/tickets/9">9 </a>for this issue</div>

**Comment by paulrosenzweig** [comment]
Fixed in #11941

**Comment by davideanastasia** [comment]
As I have mentioned in #2187 , I am experiencing the same problem. I have recovered my dashboard using the trick shown by @dacort , however I had to faff with the data in the `report_card. visualization_settings` too
